server/text_generation_server/models/__init__.py

```diff
@@ -484,10 +484,14 @@ def get_model(
         )
     sliding_window = config_dict.get("sliding_window", -1)
 
+    is_max_input_within_sliding_window = (
+        max_input_tokens <= sliding_window if max_input_tokens is not None else False
+    )
+
     if (
         (sliding_window is not None and sliding_window != -1)
         and not SUPPORTS_WINDOWING
-        and max_input_tokens > sliding_window
+        and is_max_input_within_sliding_window
```

Comment:
Maybe I'm misreading this, but shouldn't the exception be raised when the max input is *not* inside the window size?

Reply:
Yes this line was correct before.

The only thing that needs to happen is that if `max_input_tokens <= sliding_window` we can set sliding_window to -1 and forget about the windowing.

The code changes need to be much less.

Reply:
updated to set `sliding_window` to -1 if `max_input_tokens <= sliding_window` in the latest commit